pkg/controller/datavolume-controller.go

```diff
@@ -847,10 +847,19 @@ func (r *DatavolumeReconciler) reconcileSmartClonePvc(log logr.Logger,
 				r.updateCloneStatusPhase(cdiv1.CloneScheduled, datavolume, nil, SmartClone)
 		}
 
-		if err := r.client.Create(context.TODO(), newSnapshot); err != nil {
-			if !k8serrors.IsAlreadyExists(err) {
+		targetPvc := &corev1.PersistentVolumeClaim{}
```

Comment:
Yeah I think that is good

Reply:
Actually thinking about it more, this should increase the chance of seeing this flake
because there's a bigger time window for the smart clone controller to create the PVC before the Create() call on the snapshot

Something like this:
At the point of the GET on the PVC, it is legitimately not there.
But, by the time we go through isInUse, etc the PVC already got created by the smart clone controller

Reply:
hmmm, okay but maybe we should try to come up with a solution that eliminates rather than minimizes this race condition?

There are synchronization issues between the DataVolume, Smart Clone, and Namespace transfer controllers.  In this issue for example, the smart clone controller continues to create PVCs while the namespace transfer controller deletes them:  https://bugzilla.redhat.com/show_bug.cgi?id=2057148

I'm okay proceeding with this PR as-is.  It shouldn't make things worse and the solution can be revisited in fixing the aforementioned bz.

Reply:
Yeah, we should think about maybe adding some synchronization phases. That signal to another controller hey you can do your part now, and the original controller will stop doing things once past that phase.

Reply:
> hmmm, okay but maybe we should try to come up with a solution that eliminates rather than minimizes this race condition?
> 
> There are synchronization issues between the DataVolume, Smart Clone, and Namespace transfer controllers. In this issue for example, the smart clone controller continues to create PVCs while the namespace transfer controller deletes them: https://bugzilla.redhat.com/show_bug.cgi?id=2057148
> 
> I'm okay proceeding with this PR as-is. It shouldn't make things worse and the solution can be revisited in fixing the aforementioned bz.

Yeah I agree that there is a bigger issue here, merging this as is would help a little with CI immediately
but we definitely need to address the core issue which is synchronization between the various controllers